File: Medium/C++/Minimum_Size_Subarray_Sum(209).cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int s = 0, e = 0, m = INT_MAX, sum=0;
        bool p = false;
        
        while (e < nums.size()) {
            sum += nums[e]; e++;
            
            while (sum >= target) {
                p = true;
                m = min(m, e-s);
                sum -= nums[s]; s++;
            }
        }
        
        return p ? m : 0;
    }
};

File: Medium/Java/Minimum_Size_Subarray_Sum(209).java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        if (nums == null || nums.length == 0) {
            throw new IllegalArgumentException();
        }
        int len = nums.length;
        int slow = 0;
        int sum = 0;
        int minLen = len + 1;
        for (int fast = 0; fast < len; fast++) {
            sum += nums[fast];
            while (sum >= target) {
                minLen = Math.min(minLen, fast - slow + 1);
                sum -= nums[slow++];
            }
        }
        return minLen <= len ? minLen : 0;
    }
}